paddle/fluid/pir/dialect/op_generator/op_infermeta_func_gen.py

```diff
@@ -856,26 +861,52 @@ def GenDistBranch(args, op_info):
     DebugInfoForInferSpmd("{op_name}", spmd_info);
     PADDLE_ENFORCE_EQ(spmd_info.first.size(), {input_size}u, common::errors::Unavailable(
         "Size of spmd_info.first for op[{op_name}]is unexpected."));
-    for(auto& arg_dist : spmd_info.first) {{
-        dist_operand_attrs.push_back(CvtToPirAttr(arg_dist));
-    }}
 """
     dist_branch_str += TEMPLATE.format(
         spmd_func=spmd_rule_func,
         args=', '.join(infer_spmd_args_list),
-        input_size=len(op_info.input_name_list),
+        input_size=spmd_input_value_num,
```

Comment:
why not keep the constrain which force the require of infermate and inferspmd's input signature ?

the inferspmd of stack_grad could have redundant input to meet the above constrain.

Reply:
The input args of infer_spmd is consistent with infer_meta. The problem is that the auto parallel part in PIR has not handled the case when the infer_meta args (i.e. infer_spmd args) are inconsistent with the op input.